packages/core/src/mpi/__tests__/normalize-patient.test.ts

```diff
@@ -0,0 +1,159 @@
+import { PatientMPI } from "../shared";
+import { normalizePatient } from "../normalize-patient";
+import { USState } from "../../domain/geographic-locations";
+
+describe("normalizePatient", () => {
+  // Should return the same patient data if no normalization is needed
+  it("should lowercase name and address and alter street suffix", () => {
+    const PatientMPI: PatientMPI = {
+      id: "123456789",
+      firstName: "John",
+      lastName: "Dogo",
+      dob: "1990-01-01",
+      genderAtBirth: "M",
+      address: [
+        {
+          addressLine1: "123 Elm Street",
+          city: "New York",
+          zip: "10001",
+          state: USState.NY,
+        },
+      ],
+      contact: [
+        {
+          email: "john.doe@example.com",
+          phone: "123-456-7890",
+        },
+      ],
+    };
+
+    const expected: PatientMPI = {
+      id: "123456789",
+      firstName: "john",
+      lastName: "dogo",
+      dob: "1990-01-01",
+      genderAtBirth: "M",
+      address: [
+        {
+          addressLine1: "123 elm street",
+          city: "new york",
+          zip: "10001",
+          state: USState.NY,
+          country: "USA",
+        },
+      ],
+      contact: [
+        {
+          email: "john.doe@example.com",
+          phone: "1234567890",
+        },
+      ],
+    };
+
+    const result = normalizePatient(PatientMPI);
+    expect(result).toEqual(expected);
+  });
+
+  it("should normalize first and last name to lowercase", () => {
+    const PatientMPI: PatientMPI = {
+      id: "123456789",
+      firstName: "John",
+      lastName: " Smith",
+      dob: "1990-01-01",
+      genderAtBirth: "M",
+      address: [
+        {
+          addressLine1: "123 elm St",
+          city: "new york",
+          zip: "10001",
+          state: USState.NY,
+        },
+      ],
+      contact: [
+        {
+          email: "john.doe@example.com",
+          phone: "123-456-7890",
+        },
+      ],
+    };
+
+    const expected: PatientMPI = {
+      id: "123456789",
+      firstName: "john",
+      lastName: "smith",
+      dob: "1990-01-01",
+      genderAtBirth: "M",
+      address: [
+        {
+          addressLine1: "123 elm st",
+          city: "new york",
+          zip: "10001",
+          state: USState.NY,
+          country: "USA",
+        },
+      ],
+      contact: [
+        {
+          email: "john.doe@example.com",
+          phone: "1234567890",
+        },
+      ],
+    };
+
+    const result = normalizePatient(PatientMPI);
```

Comment:
This is a good example of why (really) unit tests are good: we could test `normalizeEmail` here and avoid having all that code to build patients (which, by the way could come from somewhere else, like `makePatient` or `testPatientMPI`).

Then, we could have a couple of tests to validate if `normalizeEmail` is called on `normalizePatient` based on whether the the email is set or not.

That would make it much easier to build richer tests - right now we only check one email value, but what about if the email is empty? And what if its an invalid email? Spaces before and after the email? In the middle?

Reply:
ok I added those tests and refactored. I will can add even more testing to the normalize patient framework down the line and added it to the ticket, but for now not going to do anymore